querybook/server/logic/data_element.py

```diff
@@ -0,0 +1,156 @@
+from app.db import with_session
+from const.metastore import (
+    DataElementAssociationProperty,
+    DataElementAssociationTuple,
+    DataElementAssociationType,
+    DataElementTuple,
+)
+from lib.logger import get_logger
+from logic.user import get_user_by_name
+from models.data_element import DataElement, DataElementAssociation
+
+LOG = get_logger(__file__)
+
+
+@with_session
+def get_data_element_by_id(id: int, session=None):
+    return DataElement.get(id=id, session=session)
+
+
+@with_session
+def get_data_element_by_name(name: str, session=None):
+    return session.query(DataElement).filter(DataElement.name == name).first()
+
+
+@with_session
+def get_data_element_by_column_id(column_id: int, session=None):
+    results = (
+        session.query(DataElementAssociation)
+        .filter(DataElementAssociation.column_id == column_id)
+        .all()
+    )
+
+    if not results:
+        return None
+
+    # check if there are more than 1 association type
+    association_types = set([r.type for r in results])
+    if len(association_types) > 1:
+        LOG.error(
+            f"Column {column_id} has more than one data element associated with it"
+        )
+        return None
+
+    resp = {}
+    for row in results:
+        resp["type"] = row.type.value
+        resp[row.property_name.value] = (
+            row.data_element.to_dict() if row.data_element else row.primitive_type
+        )
+    return resp
+
+
+@with_session
+def create_or_update_data_element(
```

Comment:
this function is not called anywhere?

Reply:
not used in this PR, but will be used by the data element sync task